File: Team_Workspace/Fatma_Ahmed/cpp_tasks/task6/task6.cpp
// task6.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <iostream>
using namespace std;


class Thermostat {
private: double target_temp;
public: 
    Thermostat() {};
    void setTempertaure(double temp) {
        if (temp >= 100)
        {
            cout << "WARNING:Too Hot!!" << endl;
            target_temp = 100.0;

        }
        else {
            target_temp = temp;

        }
    }
    double getTemperature() {
        return target_temp;
    }
};

int main()
{
    Thermostat cpu_monitor;
    cpu_monitor.setTempertaure(150.0);
    cout << "temperature: " << cpu_monitor.getTemperature() << endl;

}

File: Team_Workspace/dalia_ahmed/task6.py

class thermostat:
    def _init_(self):
        self._target_temp=0
    def set_temperature(self, temp):
        if temp>100:
            print("WARNING: Too Hot!")
            self._target_temp=100
        else:
            self._target_temp=temp
    def get_temperature(self):
        return self._target_temp
    
cpu_monitor=thermostat()
cpu_monitor.set_temperature(150)
print("final temp: ",cpu_monitor.get_temperature())